core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSource.scala

```diff
@@ -107,6 +108,11 @@ trait DeltaSourceBase extends Source
 
   protected var lastOffsetForTriggerAvailableNow: DeltaSourceOffset = _
 
+  override def prepareForTriggerAvailableNow(): Unit = {
+    lastOffsetForTriggerAvailableNow = DeltaSourceOffset(deltaLog.tableId,
```

Comment:
nit: there is a tableId field already.
```suggestion
    lastOffsetForTriggerAvailableNow = DeltaSourceOffset(tableId,
```